File: api/DTOs/AggregatedStatsDto.cs
namespace api.DTOs;

public class AggregatedStatsDto
{
    public List<DailyGlobalDto> GlobalDailyOrders { get; set; } = new();
    public List<DailyGlobalDto> GlobalDailyRevenue { get; set; } = new();
    public List<PerUserDailyDto> PerUserDailyOrders { get; set; } = new();
    public List<PerUserDailyDto> PerUserDailyRevenue { get; set; } = new();
    public List<PerUserMonthlyDto> PerUserMonthlyOrders { get; set; } = new();
    public List<PerUserMonthlyDto> PerUserMonthlyRevenue { get; set; } = new();
}

File: client/src/app/_models/aggregatedStatsDto.ts
import { DailyGlobalDto } from "./dailyGlobalDto";
import { PerUserDailyDto } from "./perUserDailyDto";
import { PerUserMonthlyDto } from "./perUserMonthlyDto";

export interface AggregatedStatsDto {
    globalDailyOrders: DailyGlobalDto[],
    globalDailyRevenue: DailyGlobalDto[],
    perUserDailyOrders: PerUserDailyDto[],
    perUserDailyRevenue: PerUserDailyDto[],
    perUserMonthlyOrders: PerUserMonthlyDto[],
    perUserMonthlyRevenue: PerUserMonthlyDto[]
}